packages/remix-react/components.tsx

```diff
@@ -1309,3 +1310,18 @@ function useComposedRefs<RefValueType = any>(
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, refs);
 }
+
+let OutletContext = React.createContext<AppData>(null);
+
+export function Outlet<Data = AppData>({ context }: { context?: Data }) {
```

Comment:
```suggestion
export function Outlet<Data extends AppData = AppData>({ context }: { context?: Data }) {
```
I think this should extend AppData too, that way it will match the type the OutletContext expects always (I know AppData is any but it may not be any in the future)